python/cuml/cuml/dask/common/base.py

```diff
@@ -304,9 +305,11 @@ def _run_parallel_func(
         if output_collection_type is None:
             output_collection_type = self.datatype
 
-        model_delayed = dask.delayed(
-            self._get_internal_model(), pure=True, traverse=False
-        )
+        model_delayed = self._get_internal_model()
+        if not isinstance(model_delayed, (Future, Delayed)):
+            model_delayed = dask.delayed(
+                model_delayed, pure=True, traverse=False
+            )
```

Comment:
That seems like a somewhat inefficient implementation. Can we instead modify `_get_internal_model()` to always return a Future or Delayed?